addons/NoCampaignMusic/NoCampaignMusic.lua

```diff
@@ -0,0 +1,162 @@
+zone_music_map[84] = { 252, 252, solo_id, party_id } --Batallia Downs [S]
+zone_music_map[85] = { 44, 44, solo_dungeon_id, party_dungeon_id } --La Vaule [S]
+zone_music_map[87] = { 180, 180, solo_id, party_id } --Bastok Markets [S]
+zone_music_map[88] = { 253, 253, solo_id, party_id } --North Gustaberg [S]
+zone_music_map[89] = { 0, 0, solo_id, party_id } --Grauberg [S]
+zone_music_map[90] = { 0, 0, solo_id, party_id } --Pashhow Marshlands [S]
+zone_music_map[91] = { 252, 252, solo_id, party_id } --Rolanberry Fields [S]
+zone_music_map[92] = { 44, 44, solo_dungeon_id, party_dungeon_id } --Beadeaux [S]
+zone_music_map[94] = { 182, 182, solo_id, party_id } --Windurst Waters [S]
+zone_music_map[95] = { 141, 141, solo_id, party_id } --West Sarutabaruta [S]
+zone_music_map[96] = { 0, 0, solo_id, party_id } --Fort Karugo-Narugo [S]
+zone_music_map[97] = { 0, 0, solo_id, party_id } --Meriphataud Mountains [S]
+zone_music_map[98] = { 252, 252, solo_id, party_id } --Sauromugue Champaign [S]
+zone_music_map[99] = { 44, 44, solo_dungeon_id, party_dungeon_id } --Castle Oztroja [S]
+zone_music_map[136] = { 0, 0, solo_id, party_id } --Beaucedine Glacier [S]
+zone_music_map[137] = { 42, 42, solo_id, party_id } --Xarcabard [S]
+zone_music_map[138] = { 43, 43, solo_dungeon_id, party_dungeon_id } --Castle Zvahl Baileys [S]
+zone_music_map[155] = { 43, 43, solo_dungeon_id, party_dungeon_id } --Castle Zvahl Keep [S]
+zone_music_map[164] = { 0, 0, solo_dungeon_id, party_dungeon_id } --Garlaige Citadel [S]
+zone_music_map[171] = { 0, 0, solo_dungeon_id, party_dungeon_id } --Crawlers' Nest [S]
+zone_music_map[175] = { 0, 0, solo_dungeon_id, party_dungeon_id } --The Eldieme Necropolis [S]
+
+windower.register_event('incoming chunk', function(id, data)
+    if id == 0x00A then --Zone update (zoned in)
+        local parsed = packets.parse('incoming', data)
+        if parsed['Day Music'] == campaign_id and zone_music_map[parsed.Zone] then
+            campaign_active = true
+            if not settings.active then return end
+
+            parsed['Day Music'] = zone_music_map[parsed.Zone][1]
+            parsed['Night Music'] = zone_music_map[parsed.Zone][1]
+            parsed['Solo Combat Music'] = zone_music_map[parsed.Zone][2]
+            parsed['Party Combat Music'] = zone_music_map[parsed.Zone][3]
+
+            windower.add_to_chat(8, 'Prevented campaign music.')
+            return packets.build(parsed)
+        end
+    elseif id == 0x05F then --Music update (campaign possibly started)
+        local parsed = packets.parse('incoming', data)
+        if parsed['Song ID'] == campaign_id then
+            campaign_active = true
+            if not settings.active then return end
+
+            local info = windower.ffxi.get_info()
+            if not zone_music_map[info.zone] then return end
+
+            if parsed['BGM Type'] == 0 then --only log to the chat once
+                windower.add_to_chat(8, 'Prevented campaign music.')
```

Comment:
See above, I feel this is something users will not need to know and rather it happened in the background.

Reply:
As above, this is now behind a settings flag.